File: src/screens/HomeScreen.js
import React from 'react';
import { View, Text, StyleSheet, FlatList, Image, TouchableOpacity, Pressable } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { roomsData } from '../data/rooms';
import { colors } from '../theme/colors';

export default function HomeScreen({ navigation }) {
  const [adminTapCount, setAdminTapCount] = React.useState(0);

  const handleAdminTap = () => {
    const newCount = adminTapCount + 1;
    setAdminTapCount(newCount);
    if (newCount >= 5) {
      setAdminTapCount(0);
      navigation.navigate('AdminDashboard');
    }
  };

  React.useEffect(() => {
    if (adminTapCount > 0) {
      const timer = setTimeout(() => setAdminTapCount(0), 3000);
      return () => clearTimeout(timer);
    }
  }, [adminTapCount]);

  const renderRoom = ({ item }) => (
    <TouchableOpacity 
      style={styles.card} 
      onPress={() => navigation.navigate('RoomDetails', { room: item })}
    >
      <Image source={item.images[0]} style={styles.image} />
      <View style={styles.cardContent}>
        <Text style={styles.roomName}>{item.name}</Text>
        <Text style={styles.price}>৳{item.price.toLocaleString()} <Text style={styles.perNight}>/ night</Text></Text>
        <Text style={styles.features} numberOfLines={1}>{item.features.join(' • ')}</Text>
      </View>
    </TouchableOpacity>
  );

  return (
    <SafeAreaView style={styles.container}>
      <Pressable onPress={handleAdminTap} style={styles.header}>
        <Text style={styles.headerSubtitle}>WELCOME TO</Text>
        <Text style={styles.headerTitle}>MERMAID RESORT</Text>
      </Pressable>
      
      <FlatList
        data={roomsData}
        keyExtractor={(item) => item.name}
        renderItem={renderRoom}
        contentContainerStyle={styles.listContainer}
        showsVerticalScrollIndicator={false}
      />
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1, backgroundColor: colors.background },
  header: { padding: 20, paddingTop: 10, alignItems: 'center', marginBottom: 10 },
  headerSubtitle: { fontSize: 12, color: colors.primary, letterSpacing: 4, fontWeight: 'bold', marginBottom: 4 },
  headerTitle: { fontSize: 24, fontWeight: 'bold', color: colors.text, letterSpacing: 2 },
  listContainer: { paddingHorizontal: 15, paddingBottom: 30 },
  card: { backgroundColor: colors.surface, borderRadius: 4, marginBottom: 20, overflow: 'hidden', borderWidth: 1, borderColor: colors.border },
  image: { width: '100%', height: 220, resizeMode: 'cover' },
  cardContent: { padding: 15 },
  roomName: { fontSize: 20, fontWeight: 'bold', color: colors.text, marginBottom: 6 },
  price: { fontSize: 18, color: colors.primary, fontWeight: 'bold', marginBottom: 8, fontFamily: 'monospace' },
  perNight: { fontSize: 14, color: colors.textSubtle, fontWeight: 'normal' },
  features: { fontSize: 12, color: colors.textMuted }
});

File: lib/screens/home_screen.dart
import 'package:flutter/material.dart';
import '../data/rooms.dart';
import 'room_details_screen.dart';
import 'admin_dashboard_screen.dart';

class HomeScreen extends StatelessWidget {
  final VoidCallback toggleTheme;
  final bool isDark;

  const HomeScreen({
    super.key,
    required this.toggleTheme,
    required this.isDark,
  });

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Mermaid Resort'),
        leading: IconButton(
          icon: Icon(isDark ? Icons.light_mode : Icons.dark_mode),
          onPressed: toggleTheme,
        ),
        actions: [
          IconButton(
            icon: const Icon(Icons.admin_panel_settings),
            onPressed: () {
              Navigator.push(
                context,
                MaterialPageRoute(builder: (context) => const AdminDashboardScreen()),
              );
            },
          ),
        ],
      ),
      body: ListView.builder(
        padding: const EdgeInsets.all(16),
        itemCount: roomsData.length,
        itemBuilder: (context, index) {
          final room = roomsData[index];
          return Card(
            margin: const EdgeInsets.only(bottom: 16),
            shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(12)),
            clipBehavior: Clip.antiAlias,
            child: InkWell(
              onTap: () {
                Navigator.push(
                  context,
                  MaterialPageRoute(builder: (context) => RoomDetailsScreen(room: room)),
                );
              },
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Image.asset(
                    room.images[0],
                    height: 200,
                    width: double.infinity,
                    fit: BoxFit.cover,
                    errorBuilder: (context, error, stackTrace) =>
                        Container(height: 200, color: Theme.of(context).cardColor, child: const Center(child: Icon(Icons.error))),
                  ),
                  Padding(
                    padding: const EdgeInsets.all(16),
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        Row(
                          mainAxisAlignment: MainAxisAlignment.spaceBetween,
                          children: [
                            Expanded(
                              child: Text(
                                room.name,
                                style: const TextStyle(fontSize: 20, fontWeight: FontWeight.bold),
                              ),
                            ),
                            Text(
                              '৳${room.price}',
                              style: const TextStyle(fontSize: 18, color: Color(0xFFEF4444), fontWeight: FontWeight.bold),
                            ),
                          ],
                        ),
                        const SizedBox(height: 8),
                        Wrap(
                          spacing: 8,
                          runSpacing: 8,
                          children: room.features.map((feature) => _buildFeatureChip(context, feature)).toList(),
                        ),
                      ],
                    ),
                  ),
                ],
              ),
            ),
          );
        },
      ),
    );
  }

  Widget _buildFeatureChip(BuildContext context, String label) {
    final isDarkMode = Theme.of(context).brightness == Brightness.dark;
    return Container(
      padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 4),
      decoration: BoxDecoration(
        color: isDarkMode ? const Color(0xFF222222) : Colors.grey[200],
        borderRadius: BorderRadius.circular(16),
        border: Border.all(color: isDarkMode ? const Color(0xFF333333) : Colors.grey[300]!),
      ),
      child: Text(
        label,
        style: TextStyle(fontSize: 12, color: isDarkMode ? Colors.white70 : Colors.black87),
      ),
    );
  }
}
